### PR TITLE
feat: add multi-chain support to account configs

### PR DESCRIPTION
**Pull request by moldy530**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR adds functionality related to chain management in Alchemy, including `useChain` hook, chain configuration, and chain-specific actions.

### Detailed summary
- Added `useChain` hook for managing the connected chain in React components
- Introduced chain configuration options for multiple chains support
- Implemented actions for getting, setting, and watching the active chain
- Updated store types to include chain and connections information
- Added error handling for chain not found scenarios

> The following files were skipped due to too many changes: `packages/alchemy/src/config/store/core.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/alchemy/src/config/actions/getAccount.ts [hunk 1]
```diff
@@ -15,7 +14,10 @@ export const getAccount = <TAccount extends SupportedAccountTypes>(
 ): GetAccountResult<TAccount> => {
   const accounts = config.clientStore.getState().accounts;
   if (!accounts) {
-    throw new ClientOnlyPropertyError("account");
```
Comment (moldy530) [inline packages/alchemy/src/config/actions/getAccount.ts:18]: this throws in SSR without the SSR flag on, but I think if this gets called on the server it's fine to return a disconnected state

File: packages/alchemy/src/config/actions/setChain.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { ChainNotFoundError } from "../errors.js";
+import type { AlchemyAccountsConfig } from "../types";
+
+/**
+ * Allows you to change the current chain in the core store
+ *
+ * @param config the accounts config object
+ * @param chain the chain to change to. It must be present in the connections config object
+ */
+export function setChain(config: AlchemyAccountsConfig, chain: Chain) {
```
Comment (moldy530) [inline packages/alchemy/src/config/actions/setChain.ts:11]: ah wait, this needs to be async

File: packages/alchemy/src/config/store/core.ts [hunk 3]
```diff
@@ -1,30 +1,116 @@
...
+  coreStore.subscribe(
+    ({ chain }) => chain,
+    (chain) => {
+      coreStore.setState({
+        bundlerClient: createAlchemyPublicRpcClient({
+          chain,
+          connectionConfig: coreStore.getState().connections.get(chain.id)!,
+        }),
+      });
+    }
```
Comment (moldy530) [inline packages/alchemy/src/config/store/core.ts:87]: also gonna do this as part of `setChain` so that these things get set at the same time

**Review by dphilipson — DISMISSED** [review]
Nice!

**Pull request merged**